deps/build.jl

```diff
@@ -20,36 +92,51 @@ libhealpix_src_directory = joinpath(BinDeps.depsdir(libchealpix), "src", "Healpi
-              GetSources(libchealpix)
-              @build_steps begin
-                  ChangeDirectory(joinpath(libhealpix_src_directory, "src", "C", "autotools"))
-                  `autoreconf --install`
-                  `./configure --prefix=$(usrdir(libchealpix))`
-                  `make install`
-              end
-          end), libchealpix)
+if has_libtool
+    provides(SimpleBuild,
+             (@build_steps begin
+                  GetSources(libchealpix)
+                  @build_steps begin
+                      ChangeDirectory(joinpath(libhealpix_src_directory, "src", "C", "autotools"))
+                      setenv(`autoreconf --install`, env)
+                      setenv(`./configure --prefix=$usr`, env)
+                      # Note to self: I have no idea why I need the extra \$ here and not for
+                      # libhealpix_cxx.so below. It appears to be necessary though.
+                      setenv(`make install LDFLAGS='-Wl,-rpath,\$$ORIGIN'`, env)
+                  end
+              end), libchealpix)
 
-provides(SimpleBuild,
-         (@build_steps begin
-              GetSources(libchealpix)
-              @build_steps begin
-                  ChangeDirectory(joinpath(libhealpix_src_directory, "src", "cxx", "autotools"))
-                  `autoreconf --install`
-                  `./configure --prefix=$(usrdir(libchealpix))`
-                  `make install`
-              end
-          end), libhealpix_cxx)
+    provides(SimpleBuild,
+             (@build_steps begin
+                  GetSources(libchealpix)
+                  @build_steps begin
+                      ChangeDirectory(joinpath(libhealpix_src_directory, "src", "cxx", "autotools"))
+                      setenv(`autoreconf --install`, env)
+                      setenv(`./configure --prefix=$usr`, env)
+                      setenv(`make install LDFLAGS="-Wl,-rpath,$ORIGIN"`, env)
+                  end
+              end), libhealpix_cxx)
+end
 
 libhealpixwrapper_src_directory = joinpath(BinDeps.depsdir(libhealpixwrapper), "src", "wrapper")
 
-provides(SimpleBuild,
-         (@build_steps begin
-              ChangeDirectory(libhealpixwrapper_src_directory)
-              `make`
-              `make install`
-          end), libhealpixwrapper)
+if has_pkg_config
+    provides(SimpleBuild,
+             (@build_steps begin
+                  ChangeDirectory(libhealpixwrapper_src_directory)
+                  `make PKG_CONFIG_PATH=$pkg_config_path`
+                  `make install`
+              end), libhealpixwrapper)
+end
+
+provides(Binaries,
+         URI("https://dl.bintray.com/mweastwood/LibHealpix.jl/dependencies-v0.2.1-0.tar.gz"),
+         [libcfitsio, libchealpix, libhealpix_cxx, libhealpixwrapper],
+         SHA="c5ac81b6895d567081cfecc0e29567ff9c484a64c1448f7aaca7842e390cefff",
+         os=:Linux)
+
+# https://github.com/JuliaLang/BinDeps.jl/pull/163
+is_linux() && push!(BinDeps.defaults, BinDeps.Binaries)
```

Comment:
fixed. I'm running out of steam for now though, so I'm going to leave x86 support as a todo item and just restrict the use of the binaries to x64.